base/file.jl

```diff
@@ -416,18 +416,36 @@ function touch(path::AbstractString)
     path
 end
 
-const temp_prefix = "jl_"
-
-if Sys.iswindows()
+"""
+    tempdir()
 
+Obtain the path of a temporary directory (possibly shared with other processes). On Windows,
+`tempdir()` uses the first environment variable found in the ordered list `TMP`, `TEMP`,
+`USERPROFILE`. On all other operating systems, `tempdir()` uses the first environment
+variable found in the ordered list `TMPDIR`, `TMP`, `TEMP`, and `TEMPDIR`. If none of these are
+found, the path `"/tmp"` is used.
+"""
 function tempdir()
-    temppath = Vector{UInt16}(undef, 32767)
-    lentemppath = ccall(:GetTempPathW, stdcall, UInt32, (UInt32, Ptr{UInt16}), length(temppath), temppath)
-    windowserror("GetTempPath", lentemppath >= length(temppath) || lentemppath == 0)
-    resize!(temppath, lentemppath)
-    return transcode(String, temppath)
+    path_max = 1024
+    buf = Base.StringVector(path_max)
```

Comment:
needs to be length `path_max+1` since libuv copies the trailing NUL char: https://github.com/JuliaLang/libuv/blob/26dbe5672c33fc885462c509fe2a9b36f35866fd/src/unix/core.c#L1105-L1106

Reply:
(either that or set `sz` to `path_max` … it's okay if its too small since we reallocate in the unlikely event of such a long path.)

Reply:
Thanks.
I removed the `sz` initialization, since it's not needed.